platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java

```diff
@@ -154,4 +131,74 @@ Properties loadProperties(String location) {
         return PropertiesLoader.loadProperties(location);
     }
 
+    private class CrlRefresh implements Runnable {
+
+        private Calendar start = Calendar.getInstance();
+
+        public void run() {
+            Properties encryptionProperties = loadProperties(encryptionPropertiesLocation);
+            CRL crl = crlCache.get();
+            if (crl instanceof X509CRL) {
+                X509CRL x509Crl = (X509CRL) crl;
+                if (1 == Calendar.getInstance()
+                        .getTime()
+                        .compareTo(x509Crl.getNextUpdate())) {
+                    setCrlLocation(encryptionProperties.getProperty(CRL_PROPERTY_KEY));
+                }
+            } else if (crl == null || rolledOverDay()) {
+                setCrlLocation(encryptionProperties.getProperty(CRL_PROPERTY_KEY));
+                start = Calendar.getInstance();
+            }
+        }
+
+        /**
+         * Sets the location of the CRL. Enables CRL checking if property is set, disables it otherwise
+         *
+         * @param location Location of the DER-encoded CRL file that should be used to
+         *                 check certificate revocation.
+         */
+        public void setCrlLocation(String location) {
+            CRL current = crlCache.get();
+            if (location == null) {
+                LOGGER.info("CRL property in {} is not set. Certs will not be checked against a CRL",
+                        encryptionPropertiesLocation);
+                crlCache.compareAndSet(current, null);
+            } else {
+                crlCache.compareAndSet(current, createCrl(location));
```

Comment:
❓ Should we prevent nulling out the CRL if we had a value before but got a null back from `createCrl`? @stustison what do you think about that?